### PR TITLE
feat(setup): Change default wheel to only support `cpu` builds.

### LINKED ISSUE #325: Update primeqa setup.py to build `cpu` only wheel via default install target
- [X] Change default wheel to install `CPU` only dependencies
- [X] Add extra install target as `gpu` which will install `gpu` specific python dependencies

**Solution**:
1. Default `install` target only install `faiss-cpu` python dependency.
2. Extra `gpu` target additionally installs `faise-gpu` python dependency.

To install `gpu` version:
```
pip install primeqa[gpu]
```

### PR DESCRIPTION
**Pull request by kpfadnis**
Closes #325

# Pull Request      

## What does this PR do?

```Closes #325 ```       

Notes:
- Replace `(issue)` **above** ↑↑↑ with the issue this PR closes to automatically link the two.
This must be done when the PR is created.
- Add multiple `Closes #(issue)` as needed.
- If this PR is work towards but does not close an issue, simply tag the issue without mentioning `Closes`.

## Description

**Describe** the changes proposed by this PR below to give the reviewer context below ↓↓↓

- Current PrimeQA wheel packages `gpu` specific python dependencies. E.g `faiss-gpu`
- This causes PrimeQA wheel installation on `CPU` only machine to fail.
- Updated setup.py to build `cpu` friendly wheel by default and added extra target `gpu` which allow user to install `gpu` friendly package. 

### Request Review

Be sure to **request a review** from one or more reviewers (unless the PR is to an unprotected branch).

## Versioning

When opening a PR to make changes to PrimeQA (i.e. `primeqa/`) master, be sure to increment the version following
[semantic versioning](https://semver.org/).  The VERSION is stored [here](https://github.com/primeqa/primeqa/blob/main/VERSION) 
and is incremented using `bump2version {patch,minor,major}` as described in the (development guide documentation)[development.html].

 - [X] Have you updated the VERSION?
 - [ ] Or does this PR not change the `primeqa` package or was not into master?

After pulling in changes from master to an existing PR, ensure the VERSION is updated appropriately.
This may require bumping the version again if it has been previously bumped.

If you're not quite ready yet to post a PR for review, feel free to open a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

### After Merging
If merging into master and VERSION was updated, after this PR is merged:
- [X] [Create a release](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release) from the master with version equal to [VERSION](https://github.com/primeqa/primeqa/blob/main/VERSION)
- [ ] Not merging to master or VERSION not updated

### Checklist

Review the following and mark as completed:
- [X] [Tag an issue or issues](#what-does-this-pr-do) this PR addresses.
- [X] Added [description](#description) of changes proposed.
- [X] Review requested as [appropriate](#request-review).
- [X] Version bumped as [appropriate](#Versioning).
- [ ] New classes, methods, and functions documented.
- [X] Documentation for modified code is updated.
- [ ] Built documentation to confirm it renders as expected (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Tests added to ensure all functionalities tested at >= 60% unit test coverage (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Release created [as needed](#after-merging) after merging.

### CODE REVIEW

**Comment by kpfadnis** [comment]
@bhavani105 @avisil @franzmpub Let me know if this PR looks good. Like to get it merged asap to move to `services` related PR.

File: README.md [hunk 1]
```diff
@@ -58,6 +58,9 @@ pip install 'torch~=1.11.0' --extra-index-url https://download.pytorch.org/whl/c
 # Minimal install (non-editable)
 pip install .
 
+# GPU support
```
Comment (bhavani105) [inline README.md:61]: Minimal install with GPU support
```suggestion
# GPU support
```

File: setup.py [hunk 2]
```diff
@@ -1,139 +1,132 @@
...
-    install_requires.append("torch-scatter @ https://data.pyg.org/whl/torch-1.11.0%2Bcu113/torch_scatter-2.0.9-cp39-cp39-linux_x86_64.whl")
-"""
-
 setup(
-    name="primeqa",
+    name=package_name,
+    url="https://github.com/primeqa/primeqa",
     version=version,
-    author=", ".join(authors),
+    author="PrimeQA Team",
```
Comment (bhavani105) [inline setup.py:117]: @avisil Please check the author list change from individual names to generic PrimeQA Team

File: setup.py [hunk 3]
```diff
@@ -1,139 +1,132 @@
...
     author_email="primeqa@us.ibm.com",
     description="State-of-the-art Question Answering",
     long_description=long_description,
     long_description_content_type="text/markdown",
-    url="https://github.com/primeqa/primeqa",
     license="Apache",
-    keywords=" ".join(keywords),
+    keywords="Question Answering (QA), Machine Reading Comprehension (MRC), Information Retrieval (IR)",
```
Comment (bhavani105) [inline setup.py:123]: @avisil  stripped down keywords list ok ? 
Original list keywords = [
    "NLP", "transformers", "QA", "question", "answering", "mrc", "rc", "machine", "reading", "comprehension",
    "IR", "information", "retrieval", "deep", "learning", "pytorch", "BERT", "RoBERTa", "T5", "generation", "table"
]

File: setup.py [hunk 4]
```diff
@@ -1,139 +1,132 @@
...
-
-"""
-if python_version == (3,7):
-    install_requires.append("torch-scatter @ https://data.pyg.org/whl/torch-1.11.0%2Bcu113/torch_scatter-2.0.9-cp37-cp37m-linux_x86_64.whl")
-elif python_version == (3,8):
-    install_requires.append("torch-scatter @ https://data.pyg.org/whl/torch-1.11.0%2Bcu113/torch_scatter-2.0.9-cp38-cp38-linux_x86_64.whl")
-elif python_version == (3,9):
-    install_requires.append("torch-scatter @ https://data.pyg.org/whl/torch-1.11.0%2Bcu113/torch_scatter-2.0.9-cp39-cp39-linux_x86_64.whl")
-"""
-
```
Comment (bhavani105) [inline setup.py:120]: @jdpsen Is it ok to take this out ? Should this be added to the README instead

**Review by bhavani105 — APPROVED** [review]
LGTM

Test on Linux cpu and gpu
- ran pip install -e .[all]
- verified tests run with no errors

**Review by franzmpub — APPROVED** [review]

**Pull request merged**